libsplinter/src/circuit/template/mod.rs

```diff
@@ -17,14 +17,96 @@ mod rules;
 mod yaml_parser;
 
 use std::convert::TryFrom;
+use std::path::Path;
 
 pub use error::CircuitTemplateError;
 
-use rules::{RuleArgument, Rules};
+pub use rules::RuleArgument;
+use rules::Rules;
+
 use yaml_parser::{v1, CircuitTemplate};
 
 pub(self) use crate::admin::messages::{CreateCircuitBuilder, SplinterServiceBuilder};
 
+pub const DEFAULT_TEMPLATE_DIR: &str = "/usr/share/splinter/circuit-templates";
+
+pub struct CircuitTemplateManager {
+    path: String,
+}
+
+impl Default for CircuitTemplateManager {
+    fn default() -> Self {
+        CircuitTemplateManager {
+            path: DEFAULT_TEMPLATE_DIR.to_string(),
+        }
+    }
+}
+
+impl CircuitTemplateManager {
+    pub fn new(path: &str) -> Result<CircuitTemplateManager, CircuitTemplateError> {
+        if !Path::new(&path).is_dir() {
+            Err(CircuitTemplateError::new(&format!(
+                "{} is not a valid path",
```

Comment:
"valid directory" is probably more correct (since it could be a valid path but not a valid directory).

Reply:
updated